receiver/netflowreceiver/receiver.go

```diff
@@ -5,25 +5,164 @@ package netflowreceiver // import "github.com/open-telemetry/opentelemetry-colle
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"net"
 
+	"github.com/netsampler/goflow2/v2/decoders/netflow"
+	protoproducer "github.com/netsampler/goflow2/v2/producer/proto"
+	"github.com/netsampler/goflow2/v2/utils"
+	"github.com/netsampler/goflow2/v2/utils/debug"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
 	"go.uber.org/zap"
 )
 
+var _ utils.ReceiverCallback = (*dropHandler)(nil)
+
+type dropHandler struct {
+	logger *zap.Logger
+}
+
+func (d dropHandler) Dropped(msg utils.Message) {
+	d.logger.Warn("Dropped netflow message", zap.Any("msg", msg))
+}
+
 type netflowReceiver struct {
-	// host        component.Host
-	// cancel      context.CancelFunc
-	config      *Config
-	logConsumer consumer.Logs
+	config      Config
 	logger      *zap.Logger
-	// listeners   []*Listener
+	udpReceiver *utils.UDPReceiver
+	logConsumer consumer.Logs
+}
+
+func newNetflowLogsReceiver(params receiver.Settings, cfg Config, consumer consumer.Logs) (receiver.Logs, error) {
+	// UDP receiver configuration
+	udpCfg := &utils.UDPReceiverConfig{
+		Sockets:   cfg.Sockets,
+		Workers:   cfg.Workers,
+		QueueSize: cfg.QueueSize,
+		Blocking:  false,
+		ReceiverCallback: &dropHandler{
+			logger: params.Logger,
+		},
+	}
+	udpReceiver, err := utils.NewUDPReceiver(udpCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	nr := &netflowReceiver{
+		logger:      params.Logger,
+		config:      cfg,
+		logConsumer: consumer,
+		udpReceiver: udpReceiver,
+	}
+
+	return nr, nil
 }
 
 func (nr *netflowReceiver) Start(_ context.Context, _ component.Host) error {
+	nr.logger.Info("NetFlow receiver is starting...")
+
+	// The function that will decode packets
+	decodeFunc, err := nr.buildDecodeFunc()
+	if err != nil {
+		return err
+	}
+
+	nr.logger.Info("Start listening over UDP", zap.String("scheme", nr.config.Scheme), zap.Int("port", nr.config.Port))
```

Comment:
```suggestion
	nr.logger.Info("Starting UDP listener", zap.String("scheme", nr.config.Scheme), zap.Int("port", nr.config.Port))
```
Minor wording suggestion